localstack/services/apigateway/helpers.py

```diff
@@ -584,8 +586,9 @@ def create_authorizer(path_payload: dict) -> Authorizer:
 
                     if authorizers.get(security_scheme_name):
                         return authorizers.get(security_scheme_name)
-                    authorizer = rest_api.create_authorizer(
-                        create_resource_id(),
+
+                    authorizer = Authorizer(
```

Comment:
This is really a bit of an unfortunate "split-brain" situation here (as discussed). We could also avoid using the `moto` models altogether here, and use the ASF generated `Authorizer` typed dict instead. (to avoid a situation where moto's `Authorizer.__init__` may have side effects at some point in the future) 

In any case, over time we need to find a good way to handle this setup (potentially rewriting part of the core, to decouple ourselves from moto..) 👍

Reply:
Next iteration on them will do it.